upup/pkg/fi/cloudup/awsup/aws_cloud.go

```diff
@@ -768,6 +768,9 @@ func awsBuildCloudInstanceGroup(c AWSCloud, cluster *kops.Cluster, ig *kops.Inst
 
 	for _, i := range g.Instances {
 		id := aws.StringValue(i.InstanceId)
+		if instances[id] == nil {
```

Comment:
Perhaps put after the following if block, as `instances[""]` is unlikely to be non-`nil`.

It should probably be after the `LifecycleStateTerminating` check as well.

Would we want to `klog.Warningf` for this as well?

Reply:
Makes sense to move the function. Done.

This scenario is so common that I would find warnings to be noisy. It would happen consistently through a rolling update for example.

Reply:
We consistently get the "ignoring instance as it is terminating" warning during rolling updates. I consider it more informative than noisy, but to be fair it should be info instead of warning.

I suppose in this case we don't have anything informative to log. This is a semantic change: ignoring these instances will cause them to not contribute to satisfying the "enough instances in group" cluster validation check (which might be good) but it will also prevent them from being updated if they're stuck in that state (and on an old spec).